src/core/index_simple_writer.rs

```diff
@@ -0,0 +1,47 @@
+use crate::indexer::operation::AddOperation;
+use crate::indexer::segment_updater::save_metas;
+use crate::indexer::SegmentWriter;
+use crate::{Directory, Document, Index, IndexMeta, Opstamp, Segment};
+
+#[doc(hidden)]
+pub struct IndexSimpleWriter {
```

Comment:
`SingleSegmentIndexWriter` or `IndexSingleSegmentWriter` would be better I think

Reply:
I renamed it SingleSegmentIndexWriter